src/server/pfs/server/api_server.go

```diff
@@ -719,6 +719,29 @@ func (a *apiServer) CheckStorage(ctx context.Context, req *pfs.CheckStorageReque
 	}, nil
 }
 
+func (a *apiServer) PutCache(ctx context.Context, req *pfs.PutCacheRequest) (resp *types.Empty, retErr error) {
```

Comment:
This ended up being pretty clean.  I remember we talked about how complex this would be a while ago.